### PR TITLE
Merge in Development

### PR DESCRIPTION
**Pull request by Grazfather**

### CODE REVIEW

File: util/util.py [hunk 1]
```diff
@@ -202,7 +79,7 @@ def get_ctf_by_name(database, name):
     Return the matching CTF object if found, or None otherwise.
     """
     ctfs = pickle.load(open(database, "rb"))
-    for ctf in ctfs:
+    for _, ctf in ctfs.items():
```
Comment (Grazfather) [inline util/util.py]: This could be `for ctf in ctfs.values()`

File: util/util.py [hunk 2]
```diff
@@ -156,39 +19,53 @@ def load_json(string):
...
+        "o": "о",  # \xd0\xbe
+        "O": "О",  # \xd0\x9e
+        "u": "υ",  # \xcf\x85
+        "U": "υ",  # \xcf\x85
     }
 
     return ''.join([mapping[c] if c in mapping else c for c in string])
 
 
+def getDictValue(src_dict, key):
```
Comment (Grazfather) [inline util/util.py]: Let's use python namign: `get_dict_value`

**Comment by Grazfather** [comment]
Resolved the conflict, though I'd like to test it before we merge.

File: util/util.py (hunk 2 above)
Comment (Grazfather) [reply util/util.py]: This function actually just duplicates `dict.get`

**Comment by Grazfather** [comment]
LGTM after some light testing and addressing the comments.

File: handlers/syscalls_handler.py [hunk 3]
```diff
@@ -7,45 +7,41 @@
...
 
-    def parseSyscallInfo(self, syscallEntry):
+    def parse_syscall_info(self, syscall_entries):
+        """Parse syscall information."""
         msg = "```"
 
-        for entry in syscallEntry:
-            msg += "{0:15} : {1}\n".format(entry, syscallEntry[entry])
+        for entry in syscall_entries:
+            msg += "{0:15} : {1}\n".format(entry, syscall_entries[entry])
```
Comment (Corb3nik) [inline handlers/syscalls_handler.py]: I think we can omit the `0` and `1` here.
```
>>> "{0:15}".format(1)
'              1'
>>> "{:15}".format(1)
'              1'
```

File: util/util.py [hunk 4]
```diff
@@ -231,7 +96,7 @@ def get_challenge_by_channel_id(database, challenge_channel_id):
     Return the matching Challenge object if found, or None otherwise.
     """
     ctfs = pickle.load(open(database, "rb"))
-    for ctf in ctfs:
+    for _, ctf in ctfs.items():
```
Comment (Corb3nik) [inline util/util.py]: `ctfs.values()` instead? As pointed by @Grazfather.

File: util/util.py [hunk 5]
```diff
@@ -258,13 +121,29 @@ def get_challenges_for_user_id(database, user_id, ctf_channel_id):
...
+
+
 def update_challenge_name(database, challenge_channel_id, new_name):
     """
     Updates the name of the challenge with the specified challenge id
     """
     ctfs = pickle.load(open(database, "rb"))
 
-    for ctf in ctfs:
+    for _, ctf in ctfs.items():
```
Comment (Corb3nik) [inline util/util.py]: Same thing here.

**Comment by Grazfather** [comment]
Force pushed to add those fixes to my last commit.

**Pull request merged**